gpflow/models/model.py

```diff
@@ -112,22 +113,52 @@ def predict_f(
     def predict_f_samples(
         self,
         Xnew: DataPoint,
-        num_samples: int = 1,
+        num_samples: Optional[int] = 1,
         full_cov: bool = True,
         full_output_cov: bool = False,
     ) -> tf.Tensor:
         """
         Produce samples from the posterior latent function(s) at the input points.
+
+        :param Xnew: DataPoint
+            Input locations at which to draw samples
+        :param num_samples:
+            Number of samples to draw.
+            If `None`, a single sample is drawn and the return shape is [..., N, P],
+            for any positive integer the return shape contains an extra batch
+            dimension, [..., S, N, P], with S = num_samples.
+        :param full_cov:
+            If True, draw correlated samples over the inputs. Computes the Cholesky over the
+            dense covariance matrix of size [num_data, num_data].
+            If False, draw samples that are uncorrelated over the inputs.
+        :param full_output_cov:
+            If True, draw correlated samples over the outputs.
+            If False, draw samples that are uncorrelated over the outputs.
+
+        Currently, the method does not support `full_output_cov=True` and `full_cov=True`.
         """
-        mu, var = self.predict_f(Xnew, full_cov=full_cov)  # [N, P], [P, N, N]
-        num_latent_gps = var.shape[0]
-        num_elems = tf.shape(var)[1]
-        var_jitter = ops.add_to_diagonal(var, default_jitter())
-        L = tf.linalg.cholesky(var_jitter)  # [P, N, N]
-        V = tf.random.normal([num_latent_gps, num_elems, num_samples], dtype=mu.dtype)  # [P, N, S]
-        LV = L @ V  # [P, N, S]
-        mu_t = tf.linalg.adjoint(mu)  # [P, N]
-        return tf.transpose(mu_t[..., np.newaxis] + LV)  # [S, N, P]
+        if full_cov and full_output_cov:
+            raise NotImplementedError(
+                "The combination of both `full_cov` and `full_output_cov` is not supported."
+            )
+
+        mean, cov = self.predict_f(Xnew, full_cov=full_cov)  # [N, P], [P, N, N] or [N, P]
```

Comment:
should it pass full_output_cov as well?
```suggestion
        mean, cov = self.predict_f(Xnew, full_cov=full_cov, full_output_cov=full_output_cov)  # [N, P], [P, N, N] or [N, P]
```
?